Routing Zone Data Source Bug

Some instances of the `apstra_datacenter_routing_zones` Data Source causes the provider to crash. This crash does not occur in all instances of the Data Source.

The HCL code below shows the variables, resource, and data source for an instance of the `apstra_datacenter_routing_zones` Data Source that does crash.

This behaviour is seen on v0.24.0 and v0.25.0 of the provider.

---

```hcl
variable "blueprint" {
    type = object({
        name        = string
        template_id = string
    })
    description = "Apstra blueprint instatiation."
    default     = {
        name        = "edtnabtf_pod_11_test"
        template_id = "L2_Virtual_EVPN"
    }
 }

variable "routing_zones"{
    type = map(object({
        name                       = string
        vlan_id                    = number
        vni                        = number
        dhcp_servers               = list(string)
    }))
    description = "Creating a routing zone for a Blueprint within the Apstra server."
    default = {
        tf_rz_test_01 = {
            name         = "tf_rz_test_01"
            vlan_id      = "5"
            vni          = "5000"
            dhcp_servers = ["192.168.100.10","192.168.200.10"]
        }
    }
 }

resource apstra_datacenter_blueprint blueprint {
    name        = var.blueprint.name
    template_id = var.blueprint.template_id
}

data apstra_datacenter_routing_zones routing_zone {
    blueprint_id = apstra_datacenter_blueprint.blueprint.id
    for_each     = var.routing_zones
    filters = {
        name = each.value.name
    }
}
```

```sh
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
Stack trace from the terraform-provider-apstra_v0.25.0 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb8f90c]
goroutine 412 [running]:
terraform-provider-apstra/apstra/blueprint.(*DatacenterRoutingZone).LoadApiData(0xc0001a60f0?, {0xf4da50?, 0xc0004b32c0?}, 0xc0002a4b70?, 0x24?)
	terraform-provider-apstra/apstra/blueprint/datacenter_routing_zone.go:265 +0x4c
terraform-provider-apstra/apstra.(*dataSourceDatacenterRoutingZone).Read(0xc000012008, {0xf4da50, 0xc0004b32c0}, {{{{0xf526a8, 0xc00059f320}, {0xcd3ee0, 0xc00059e6c0}}, {0xf55240, 0xc000480b80}}, {{{0x0, ...}, ...}, ...}}, ...)
	terraform-provider-apstra/apstra/data_source_datacenter_routing_zone.go:89 +0x40e
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadDataSource(0xc0000d38c0, {0xf4da50, 0xc0004b32c0}, 0xc0004b39b0, 0xc000567650)
	github.com/hashicorp/terraform-plugin-framework@v1.2.0/internal/fwserver/server_readdatasource.go:75 +0x4ad
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadDataSource(0xc0000d38c0, {0xf4da50?, 0xc0004b30e0?}, 0xc0003742c0)
	github.com/hashicorp/terraform-plugin-framework@v1.2.0/internal/proto6server/server_readdatasource.go:52 +0x42f
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadDataSource(0xc0001b4aa0, {0xf4da50?, 0xc0004b2930?}, 0xc0005b4d20)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/tf6server/server.go:656 +0x403
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadDataSource_Handler({0xdc3d20?, 0xc0001b4aa0}, {0xf4da50, 0xc0004b2930}, 0xc0000c4150, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b21e0, {0xf52998, 0xc000382340}, 0xc000982000, 0xc00033cc90, 0x14c5090, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc0001b21e0, {0xf52998, 0xc000382340}, 0xc000982000, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a
Error: The terraform-provider-apstra_v0.25.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/apstra/blueprint/datacenter_routing_zone.go
+++ b/apstra/blueprint/datacenter_routing_zone.go
@@ -262,7 +262,12 @@ func (o *DatacenterRoutingZone) DhcpServerRequest(_ context.Context, _ *diag.Dia
 
 func (o *DatacenterRoutingZone) LoadApiData(_ context.Context, sz *apstra.SecurityZoneData, _ *diag.Diagnostics) {
 	o.Name = types.StringValue(sz.VrfName)
-	o.VlanId = types.Int64Value(int64(*sz.VlanId))
+
+	if sz.VlanId != nil {
+		o.VlanId = types.Int64Value(int64(*sz.VlanId))
+	} else {
+		o.VlanId = types.Int64Null()
+	}
 
 	if sz.RoutingPolicyId != "" {
 		o.RoutingPolicyId = types.StringValue(sz.RoutingPolicyId.String())

